### PR TITLE
feat: manage collections at seller level

### LINKED ISSUE #547: manage collections at seller level
- provide a service for a seller to add a new collection
- list all collections per seller in the subgraph
- list all collections properties in the subgraph (based on contractURI metadata + onchain data: collectionIndex, externalId, contract address)
- link offers to collections they belong to in the subgraph

### PR DESCRIPTION
**Pull request by levalleux-ludo**
## Description

fixes #547 

## How to test

new e2e tests `core-sdk-collections.test.ts`

### CODE REVIEW

File: packages/common/src/types/accounts.ts [hunk 1]
```diff
@@ -140,3 +138,17 @@ export type DisputeResolverStruct = {
...
+  /** A tag to identify the seller's collection */
+  collectionId: string;
+  /** Uri of the metadata of the collection voucher contract */
+  contractUri: string;
+  /** Royalty percentage for voucher secondary sales */
+  royaltyPercentage: BigNumberish;
+  /** collectionSalt is added to the seller admin address to give the sellerSalt
+      that is used to compute the voucher contract address. By default, deduced from collectionId */
+  collectionSalt?: string;
+  /** optional field to specify the sellerId (will be retrived from caller wallet is not specified) */
```
Comment (albertfolch-redeemeum) [inline packages/common/src/types/accounts.ts:152]:
```suggestion
  /** optional field to specify the sellerId (will be retrieved from caller wallet if not specified) */
```

File: packages/core-sdk/src/accounts/handler.ts [hunk 2]
```diff
@@ -26,25 +32,52 @@ import {
...
+  // In that case, we add a random suffix on the collectionId to generate a different salt
+  let collectionSalt = formatBytes32String(INITIAL_COLLECTION_ID);
+  const isAvailable = await isSellerSaltAvailable({
+    adminAddress: args.sellerToCreate.admin,
+    salt: collectionSalt,
+    contractAddress: args.contractAddress,
+    web3Lib: args.web3Lib
+  });
+  if (!isAvailable) {
+    const uuid = Math.floor(Math.random() * 100000000).toFixed(0);
```
Comment (albertfolch-redeemeum) [inline packages/core-sdk/src/accounts/handler.ts:59]:
```suggestion
    const uuid = crypto.randomUUID();
```

File: packages/subgraph/src/mappings/account-handler.ts [hunk 3]
```diff
@@ -535,3 +577,83 @@ export function handleAllowedSellersRemovedEvent(
...
+    offerCollection.collectionAddress = collectionAddress;
+    offerCollection.externalIdHash = externalIdHash;
+    offerCollection.externalId = externalId;
+    offerCollection.metadata = offerCollectionId;
+    offerCollection.save();
+  } else {
+    log.warning("Offer collection with ID '{}' already exists!", [
+      offerCollectionId
+    ]);
+  }
```
Comment (albertfolch-redeemeum) [inline packages/subgraph/src/mappings/account-handler.ts:613]: NIT: swap conditions

```suggestion
  if (offerCollection) {
     log.warning("Offer collection with ID '{}' already exists!", [
        offerCollectionId
      ]);
  } else {
    offerCollection = new OfferCollection(offerCollectionId);
    offerCollection.sellerId = sellerId;
    offerCollection.seller = sellerId.toString();
    offerCollection.collectionIndex = collectionIndex;
    offerCollection.collectionAddress = collectionAddress;
    offerCollection.externalIdHash = externalIdHash;
    offerCollection.externalId = externalId;
    offerCollection.metadata = offerCollectionId;
    offerCollection.save();
  }
```

**Review by albertfolch-redeemeum — APPROVED** [review]
LGTM

File: packages/core-sdk/src/accounts/handler.ts (hunk 2 above)
Comment (levalleux-ludo) [reply packages/core-sdk/src/accounts/handler.ts]: done

Comment (levalleux-ludo) [reply packages/core-sdk/src/accounts/handler.ts:59]: I've finally reverted as the crypto symbol is not resolved in tests...

**Pull request merged**